internal/renderers/femtovg/itemrenderer.rs

```diff
@@ -1388,69 +1389,73 @@ impl<'a> GLItemRenderer<'a> {
+                        femtovg::PixelFormat::Rgba8,
+                        femtovg::ImageFlags::PREMULTIPLIED
+                            | femtovg::ImageFlags::REPEAT_X
+                            | femtovg::ImageFlags::REPEAT_Y
+                            | scaling_flags,
+                    )
+                    .expect("internal error allocating temporary texture for image tiling");
+
+                let mut image_rect = femtovg::Path::new();
+                image_rect.rect(0., 0., texture_size.width, texture_size.height);
+                self.canvas.borrow_mut().save_with(|canvas| {
+                    canvas.reset();
+                    canvas.scale(1., -1.); // Image are rendered upside down
+                    canvas.translate(
                         0.,
-                        fit.clip_rect.width() as f32,
-                        fit.clip_rect.height() as f32,
+                        -(buf_size.height / orig_size.height) * (fit.clip_rect.height() as f32),
                     );
-                    self.canvas.borrow_mut().save_with(|canvas| {
-                        canvas.reset();
-                        canvas.scale(1., -1.); // Image are rendered upside down
-                        canvas.translate(0., -fit.clip_rect.height() as f32);
-                        canvas.set_render_target(femtovg::RenderTarget::Image(clipped_image));
-                        canvas.global_composite_operation(femtovg::CompositeOperation::Copy);
-                        canvas.fill_path(
-                            &image_rect,
-                            &femtovg::Paint::image(
-                                image_id,
-                                -fit.clip_rect.origin.x as f32,
-                                -fit.clip_rect.origin.y as f32,
-                                image_size.cast().width,
-                                image_size.cast().height,
-                                0.,
-                                1.0,
-                            ),
-                        );
-                        canvas.set_render_target(self.current_render_target());
-                    });
-                    self.textures_to_delete_after_flush
-                        .borrow_mut()
-                        .push(Texture::adopt(&self.canvas, clipped_image));
-                    (clipped_image, Default::default(), fit.clip_rect.size.cast())
-                } else {
-                    (image_id, fit.clip_rect.origin.cast::<f32>(), image_size.cast())
-                };
+                    canvas.set_render_target(femtovg::RenderTarget::Image(clipped_image));
+                    canvas.global_composite_operation(femtovg::CompositeOperation::Copy);
+                    canvas.fill_path(
+                        &image_rect,
+                        &femtovg::Paint::image(
+                            image_id,
+                            -(buf_size.width / orig_size.width) * (fit.clip_rect.origin.x as f32),
```

Comment:
buf_size / orig_size (width and height) appears a few times as the scale factor. Perhaps the code would be more readable if that factor was held in a variable similar to source_to_target. Like maybe cached_image_to_rendered?

Reply:
done